src/rebar_prv_plugins_upgrade.erl

```diff
@@ -35,14 +35,18 @@ do(State) ->
     {Args, _} = rebar_state:command_parsed_args(State),
     case proplists:get_value(plugin, Args, none) of
         none ->
-            ?PRV_ERROR(no_plugin_arg);
+            {_, LocalPluginsNames} = rebar_prv_plugins:list_local_plugins(State),
+            lists:foldl(
+                fun (LocalPluginName, {ok, StateAcc}) ->
+                    upgrade(atom_to_list(LocalPluginName), StateAcc)
```

Comment:
This is going to be rather slow or costly... you're calling the whole upgrade procedure for each plugin one at a time, including analysis. I can merge this since it's better than the current behaviour, but I'd look in seeing if we could improve this at some point by letting the code from lines 70 to 80 handle a larger set of applications in a list to build them at once.
It's not urgent since people upgrade plugins rather rarely, but the code is currently suboptimal.

Reply:
Sure, it's not urgent; and I will look into possibilities to improve it (this was mainly a first dirty approach to prove the concept). For context's sake: I've it plugged to CI, which means that I'm currently forced to know, beforehand, which plugins I want to upgrade and then do it. If it's in a generic Makefile, for example (whose consumers I know not what plugins chose), I could easily just call `rebar3 plugins upgrade` and let `rebar3` take care of it. I'll change to draft to signal it's still ongoing.

Reply:
the global profile is one magic thing that is super odd. Otherwise, the merging of state and handling of profiles should be transparent and nobody should need to care about them. It's why we aim at people not needing to know which profiles they run under.

Reply:
Cool, then I guess I might "bypass" those lines in my analysis, as you propose only looking at the 70-80 range.

Reply:
We shouldn't block this PR on these improvements, they're more of a follow-up. Having something working slow is better than something that doesn't work at all in this case.

Reply:
> I'm ready to merge this whenever you're happy to take it out of draft mode.

Oh, cool. I'm OK to do that just now.